src/Facebook/FileUpload/FacebookResumableUploader.php

```diff
@@ -0,0 +1,208 @@
+                $res['start_offset'],
+                $res['end_offset'] - $res['start_offset']
+            );
+
+            return $firstTransferChunk;
+        } catch (FacebookResponseException $e) {
+            throw new FacebookUploadException($e->getResponse());
+        }
+    }
+
+    /**
+     * Upload - phase transfer
+     *
+     * @param $endpoint
+     * @param FacebookTransferChunk $chunk
+     * @return FacebookTransferChunk
+     *
+     * @throws FacebookUploadTransferException
+     * @throws FacebookSDKException
+     */
+    public function transfer($endpoint, FacebookTransferChunk $chunk)
+    {
+        $transReqParams = [
+            'upload_phase' => 'transfer',
+            'upload_session_id' => $chunk->getUploadSessionId(),
+            'start_offset' => $chunk->getStartOffset(),
+            'video_file_chunk' => $chunk,
+        ];
+
+        $maxTransferTries = $this->maxTransferTries;
+
+        $request = new FacebookRequest(
+            $this->app,
+            $this->accessToken,
+            'POST',
+            $endpoint,
+            $transReqParams
+        );
+
+        while ($maxTransferTries > 0) {
+            try {
+                $res= $this->client->sendRequest($request)->getDecodedBody();
+
+                $nextTransferChunk = new FacebookTransferChunk(
+                    $chunk->getFilePath(),
+                    $chunk->getUploadSessionId(),
+                    $chunk->getVideoId(),
+                    $res['start_offset'],
+                    $res['end_offset'] - $res['start_offset']
+                );
+
+                return $nextTransferChunk;
+            } catch (FacebookResponseException $e) {
+                if (--$maxTransferTries <= 0) {
+                    $resumeContext = new FacebookResumeContext(
+                        $endpoint,
+                        $this->accessToken,
+                        $chunk
+                    );
+                    throw new FacebookUploadTransferException($e->getResponse(), $resumeContext);
+                }
+                usleep(500000);
+                continue;
+            }
+        }
+    }
+
+    /**
+     * Upload - phase finish
+     *
+     * @param $endpoint
+     * @param $uploadSessionId
+     * @return boolean
+     *
+     * @throws FacebookUploadException
+     * @throws FacebookSDKException
+     * @throws null
+     */
+    public function finish($endpoint, $uploadSessionId)
+    {
+        $finishReqParams = [
+            'upload_phase' => 'finish',
+            'upload_session_id' => $uploadSessionId,
+        ];
+
+        try {
+            $request = new FacebookRequest(
+                $this->app,
+                $this->accessToken,
+                'POST',
+                $endpoint,
+                $finishReqParams
+            );
+
+            $res = $this->client->sendRequest($request)->getDecodedBody();
+            return $res['success'];
+        } catch (FacebookResponseException $e) {
```

Comment:
There should be no catch here since there is not resumable errors on "finish" phase. Let the exception bubble up.